src/ObjLoading/Game/T6/AssetLoaders/AssetLoaderSoundBank.cpp

```diff
@@ -0,0 +1,584 @@
+    alias->flags.volumeGroup = GetValueIndex(row.GetValue("group"), ObjConstants::SOUND_GROUPS.data(), ObjConstants::SOUND_GROUPS.size());
+    alias->flags.fluxType = GetValueIndex(row.GetValue("move_type"), ObjConstants::SOUND_MOVE_TYPES.data(), ObjConstants::SOUND_MOVE_TYPES.size());
+    alias->flags.loadType = GetValueIndex(row.GetValue("type"), ObjConstants::SOUND_LOAD_TYPES.data(), ObjConstants::SOUND_LOAD_TYPES.size());
+    alias->flags.busType = GetValueIndex(row.GetValue("bus"), ObjConstants::SOUND_BUS_IDS.data(), ObjConstants::SOUND_BUS_IDS.size());
+    alias->flags.limitType = GetValueIndex(row.GetValue("limit_type"), ObjConstants::SOUND_LIMIT_TYPES.data(), ObjConstants::SOUND_LIMIT_TYPES.size());
+    alias->flags.volumeFalloffCurve = GetValueIndex(row.GetValue("volume_falloff_curve"), ObjConstants::SOUND_CURVES.data(), ObjConstants::SOUND_CURVES.size());
+    alias->flags.reverbFalloffCurve = GetValueIndex(row.GetValue("reverb_falloff_curve"), ObjConstants::SOUND_CURVES.data(), ObjConstants::SOUND_CURVES.size());
+
+    alias->flags.entityLimitType =
+        GetValueIndex(row.GetValue("entity_limit_type"), ObjConstants::SOUND_LIMIT_TYPES.data(), ObjConstants::SOUND_LIMIT_TYPES.size());
+    alias->flags.volumeMinFalloffCurve =
+        GetValueIndex(row.GetValue("volume_min_falloff_curve"), ObjConstants::SOUND_CURVES.data(), ObjConstants::SOUND_CURVES.size());
+    alias->flags.reverbMinFalloffCurve =
+        GetValueIndex(row.GetValue("reverb_min_falloff_curve"), ObjConstants::SOUND_CURVES.data(), ObjConstants::SOUND_CURVES.size());
+    alias->flags.randomizeType =
+        GetValueIndex(row.GetValue("randomize_type"), ObjConstants::SOUND_RANDOMIZE_TYPES.data(), ObjConstants::SOUND_RANDOMIZE_TYPES.size());
+
+    return true;
+}
+
+bool LoadSoundAliasIndexList(MemoryManager* memory, SndBank* sndBank)
+{
+    // contains a list of all the alias ids in the sound bank
+    sndBank->aliasIndex = static_cast<SndIndexEntry*>(memory->Alloc(sizeof(SndIndexEntry) * sndBank->aliasCount));
+    memset(sndBank->aliasIndex, 0xFF, sizeof(SndIndexEntry) * sndBank->aliasCount);
+
+    bool* setAliasIndexList = new bool[sndBank->aliasCount];
+    memset(setAliasIndexList, false, sndBank->aliasCount);
+
+    for (auto i = 0u; i < sndBank->aliasCount; i++)
+    {
+        auto idx = sndBank->alias[i].id % sndBank->aliasCount;
+        if (sndBank->aliasIndex[idx].value == USHRT_MAX)
+        {
+            sndBank->aliasIndex[idx].value = i;
+            sndBank->aliasIndex[idx].next = USHRT_MAX;
+            setAliasIndexList[i] = true;
+        }
+    }
+
+    for (auto i = 0u; i < sndBank->aliasCount; i++)
+    {
+        if (setAliasIndexList[i])
+            continue;
+
+        auto idx = sndBank->alias[i].id % sndBank->aliasCount;
+        while (sndBank->aliasIndex[idx].next != USHRT_MAX)
+        {
+            idx = sndBank->aliasIndex[idx].next;
+        }
+
+        auto offset = 1u;
+        auto freeIdx = USHRT_MAX;
```

Comment:
careful: `freeIdx` has type `int` here. Because USHRT_MAX is just a macro for `0xffff` which makes this an int.
If you want to do the same thing the c++ way while also being guaranteed to get the right type you can use:
```cpp
auto freeIdx = std::numeric_limits<unsigned short>::max();
```

(It's a constexpr method so it will be optimized out, dont worry)